WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -360,6 +384,11 @@ private void restoreSavedInstanceState(Bundle savedInstanceState) {
             isInSearchMode = savedInstanceState.getBoolean(KEY_IS_IN_SEARCH_MODE);
             lastSearch = savedInstanceState.getString(KEY_LAST_SEARCH);
             mSitePickerMode = (SitePickerMode) savedInstanceState.getSerializable(KEY_SITE_PICKER_MODE);
+
+            mHashSelectedPositions = (HashSet<Integer>) savedInstanceState.getSerializable(KEY_SELECTED_POSITIONS);
```

Comment:
Just a note here. We have an `unchecked cast` from `java.io.Serializable` to `java.util.HashSet<java.lang.Integer>` that could lead to a`ClassCastException` if at runtime the type cannot be found. In these situations, a `Parcelable` is normally used to avoid this but I think for this solution we can use this as-is.